libs/native-federation-runtime/src/lib/load-remote-module.ts

```diff
@@ -1,54 +1,40 @@
-/* eslint-disable @typescript-eslint/no-explicit-any */
-import { appendImportMap } from './utils/add-import-map';
 import { processRemoteInfo } from './init-federation';
-import {
-  getRemote,
-  getRemoteNameByBaseUrl,
-  isRemoteInitialized,
-} from './model/remotes';
-import { getDirectory, joinPaths } from './utils/path-utils';
-
-declare function importShim<T>(url: string): T;
+import { getDirectory } from './utils/path-utils';
+import { getRemoteNameByBaseUrl, isRemoteInitialized } from './model/remotes';
 
 export type LoadRemoteModuleOptions = {
   remoteEntry?: string;
   remoteName?: string;
   exposedModule: string;
 };
 
-export async function loadRemoteModule<T = any>(
-  options: LoadRemoteModuleOptions
-): Promise<T>;
-export async function loadRemoteModule<T = any>(
+export async function loadRemoteModule<
+  Default = any,
+  Exports extends object = any
+>(options: LoadRemoteModuleOptions): Promise<{ default: Default } & Exports>;
+export async function loadRemoteModule<
+  Default = any,
+  Exports extends object = any
+>(
   remoteName: string,
   exposedModule: string
-): Promise<T>;
-export async function loadRemoteModule<T = any>(
+): Promise<{ default: Default } & Exports>;
+export async function loadRemoteModule<
+  Default = any,
+  Exports extends object = any
+>(
   optionsOrRemoteName: LoadRemoteModuleOptions | string,
   exposedModule?: string
-): Promise<T> {
+): Promise<{ default: Default } & Exports> {
   const options = normalizeOptions(optionsOrRemoteName, exposedModule);
 
   await ensureRemoteInitialized(options);
 
   const remoteName = getRemoteNameByOptions(options);
-  const remote = getRemote(remoteName);
```

Comment:
#### Remote unnecessary global cache for loading the remotes
The `es-module-shims` is directly used to load modules instead of caching not-updated